### ISSUE TITLE
darwin/arm64 homebrew "tinygo test" on google/btree fails with "runtime: goroutine stack exceeds 1000000000-byte limit"

### ISSUE THREAD

**Issue by dkegel-fastly** [tinygo-org/tinygo#2957]
With go and tinygo from homebrew on an m1 mac, this fails:
```
% git clone https://github.com/google/btree
% cd btree
% go version
go version go1.18.3 darwin/arm64
% go test
1656943803
PASS
ok      github.com/google/btree 0.930s
% tinygo version
tinygo version 0.24.0 darwin/amd64 (using go version go1.18.3 and LLVM version 14.0.0)
% tinygo test
runtime: goroutine stack exceeds 1000000000-byte limit
runtime: sp=0xc03fe804f8 stack=[0xc03fe80000, 0xc05fe80000]
fatal error: stack overflow
```

As a sanity check, I also tried dgryski/go-arc; that passed.

**Comment by dkegel-fastly** [comment]
See analysis at https://gophers.slack.com/archives/CN4R2DQV7/p1655394158813439

**Comment by dgryski** [comment]
This seems to be working now
```
~/go/src/github.com/google/btree $ git show |head -3
commit 8e29150ba321eef204059de2ab494f179b6cff2c
Author: Graeme Connell <gsconnell@gmail.com>
Date:   Wed Jun 15 16:06:10 2022 -0600
~/go/src/github.com/google/btree $ tinygo version
tinygo version 0.26.0-dev-c199dd03 darwin/amd64 (using go version go1.19 and LLVM version 14.0.0)
~/go/src/github.com/google/btree $ tinygo test
1662141478
ok  	github.com/google/btree	3.226s
~/go/src/github.com/google/btree $ tinygo test -target=wasi
1662141548
ok  	github.com/google/btree	11.872s
```

**Comment by aykevl** [comment]
Pretty sure this was fixed by https://github.com/tinygo-org/tinygo/pull/3003.

**Comment by deadprogram** [comment]
Released as part of `v0.26.0`
